packages/jest-jasmine2/src/index.js

```diff
@@ -43,16 +43,31 @@ function jasmine2(config, environment, moduleLoader, testPath) {
       moduleLoader.requireModule(null, config.setupTestFrameworkScriptFile);
     }
   });
-
+  function isSpyLike(test) {
+    return !! (test.calls && test.calls.all);
+  }
+  function isMockLike(test) {
+    return test.mock !== undefined;
+  }
   env.beforeEach(() => {
     jasmine.addMatchers({
       toBeCalled: (/* util, customEqualityTesters */) => {
         return {
-          compare: (actual/*, expected */) => {
-            if (actual.mock === undefined) {
-              throw Error('toBeCalled() should be used on a mock function');
+          compare: (actual, expected) => {
+            if (expected) {
+              throw Error(
+                'toBeCalled() does not accept parameters, use toBeCalledWith instead'
+              );
+            }
+            if (!isMockLike(actual) && !isSpyLike(actual)) {
+              throw Error('toBeCalled() should be used on a mock function or a jasmine spy');
+            }
+            let calls = actual.mock.calls;
```

Comment:
I think I'd probably do something like this:

const isMock = isMockLike(actual);
if (!isMock && isSpyLike(actual)) {
  throw Error(…);
}

let calls = isMock ? actual.mock.calls : actual.calls.all().map(x => x.args);

overall it is shorter and you are reducing the amount of function calls to `isMockLike` and `isSpyLike`. Because you are doing `actual.mock !== undefined` in the `isMockLike` function and then just `actual.mock ? …` in the functions here, it is also slightly inconsistent. What do you think?